clients/fides-js/src/lib/consent-utils.ts

```diff
@@ -238,13 +238,17 @@ export const shouldResurfaceConsent = (
   cookie: FidesCookie,
   savedConsent: NoticeConsent,
 ): boolean => {
+  // Always resurface consent for TCF unless the saved version_hash matches
   if (experience.experience_config?.component === ComponentType.TCF_OVERLAY) {
     if (experience.meta?.version_hash) {
       return experience.meta.version_hash !== cookie.tcf_version_hash;
     }
-    // Ensure we always resurface consent for TCF if for some reason experience does not have version_hash
     return true;
   }
+  // Never surface consent for modal-only experiences
+  if (experience.experience_config?.component === ComponentType.MODAL) {
+    return false;
+  }
```

Comment:
This is the fix - it's effectively a one-line adjustment to make this utility `shouldResurfaceConsent()` behave as expected when using a modal-only experience. This method is called directly by `Fides.shouldShowExperience()` to determine whether or not the experience will be shown to the user automatically during the `init()` sequence.

(the rest of the PR is E2E tests for all the experience types)